x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/50_engine_list.yml

```diff
@@ -0,0 +1,143 @@
+setup:
+  - do:
+      indices.create:
+        index: test-index1
+        body:
+          settings:
+            index:
+              number_of_shards: 1
+              number_of_replicas: 0
+
+  - do:
+      indices.create:
+        index: test-index2
+        body:
+          settings:
+            index:
+              number_of_shards: 1
+              number_of_replicas: 0
+
+  - do:
+      engine.put:
+        engine_id: test-engine-1
+        body:
+          indices: [ "test-index1" ]
+          analytics_collection_name: "test-analytics1"
+
+  - do:
+      engine.put:
+        engine_id: test-engine-2
+        body:
+          indices: [ "test-index2" ]
+          analytics_collection_name: "test-analytics2"
+
+  - do:
+      engine.put:
+        engine_id: another-test-engine
+        body:
+          indices: [ "test-index1", "test-index2" ]
+          analytics_collection_name: "test-another-analytics"
+
+---
+teardown:
+  - do:
+      engine.delete:
+        engine_id: test-engine-1
+        ignore: 404
+
+  - do:
+      engine.delete:
+        engine_id: test-engine-2
+        ignore: 404
+
+  - do:
+      engine.delete:
+        engine_id: another-test-engine
+        ignore: 404
+
+---
+"List Engines":
```

Comment:
The Rails API limited engines to those we had permissions to test against. So if there existed a `test-engine-3` that we didn't have permissions on it wouldn't be returned in results. 

I don't _think_ we have to explicitly test that here since the index permissions should take care of it for us? But I wanted to ask just in case.

Reply:
Someone with cluster_manage_engines will be able to see all engines (list and get). When they want to update them, then index permissions will kick in.